source/Nevermore/CachingTableColumnNameResolver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Linq;
+
+namespace Nevermore
+{
+    public class CachingTableColumnNameResolver : ITableColumnNameResolver
```

Comment:
anything new that we don't _need_ to add to the public surface area of the library we should avoid doing so. Best to start with these being internal and change them later if necessary.